x-pack/plugins/security_solution/server/search_strategy/index_fields/index.ts

```diff
@@ -26,13 +26,10 @@ export const securitySolutionIndexFieldsProvider = (): ISearchStrategy<
   const beatFields: BeatFields = require('../../utils/beat_schema/fields').fieldsBeat;
 
   return {
-    search: (request, options, context) =>
+    search: ({ esClient }, request, options) =>
       from(
         new Promise<IndexFieldsStrategyResponse>(async (resolve) => {
-          const { elasticsearch } = context.core;
-          const indexPatternsFetcher = new IndexPatternsFetcher(
-            elasticsearch.legacy.client.callAsCurrentUser
-          );
+          const indexPatternsFetcher = new IndexPatternsFetcher(esClient.asCurrentUser);
```

Comment:
This is the portion that relies on https://github.com/elastic/kibana/pull/80116.